pkg/trait/pull_secret.go

```diff
@@ -110,7 +110,7 @@ func (t *pullSecretTrait) Apply(e *Environment) error {
 			})
 		})
 	}
-	if util.IsTrue(t.ImagePullerDelegation) {
+	if util.IsTrue(t.ImagePullerDelegation) && e.IntegrationInPhase(v1.IntegrationPhaseDeploying) {
```

Comment:
Actually, thinking about this, I think we should refactor the `delegateImagePuller` method:
* Add the RoleBinding to the environment resources, instead of manually creating it
* Call the `delegateImagePuller` method in both deploying and running phase

That way, we follow the idiomatic way of managing resources, and let the _deployer_ and _gc_ do their duties.

Reply:
Changes done, I found it better to remove `delegateImagePuller` method as it was not doing much.

Reply:
Thanks, I agree with you 👍🏼.